### PR TITLE
MULE-10778: Add attachment support to the new Web Service Consumer

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/MessageDispatcherInterceptor.java [hunk 1]
```diff
@@ -44,39 +51,44 @@ public MessageDispatcherInterceptor(String address, MessageObserver messageObser
...
       Message inMessage = new MessageImpl();
 
-      String encoding = "UTF-8";
-      inMessage.put(Message.ENCODING, encoding);
-
-      String contentType = "text/xml";
-      if (!contentType.contains("charset")) {
-        contentType += "; charset=" + encoding;
-      }
+      // TODO make encoding policy
```
Comment (marianogonzalez) [inline extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/MessageDispatcherInterceptor.java]: ticket

File: extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/OutputSoapAttachmentsInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  public void handleMessage(Message message) throws Fault {
+    Collection<Attachment> attachments = message.getAttachments();
+    List attachmentParts = emptyList();
+    if (attachments != null && !attachments.isEmpty()) {
+      attachmentParts = attachments.stream().map(this::toMessagePart).collect(new ImmutableListCollector<>());
+    }
+    message.getExchange().put(MULE_ATTACHMENTS_KEY, attachmentParts);
+  }
+
+  private org.mule.runtime.api.message.Message toMessagePart(Attachment attachment) {
```
Comment (marianogonzalez) [inline extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/OutputSoapAttachmentsInterceptor.java]: short name

File: extensions/wsc/src/main/java/org/mule/extension/ws/internal/WscConnectionProvider.java [hunk 3]
```diff
@@ -47,6 +47,13 @@
   private String address;
 
   /**
+   * If should use the MTOM protocol to manage the attachments or not.
+   */
+  @Parameter
+  @Optional(defaultValue = "false")
+  private boolean mtomEnabled;
```
Comment (marianogonzalez) [inline extensions/wsc/src/main/java/org/mule/extension/ws/internal/WscConnectionProvider.java]: is there a use case for wanting to set this on the request level?

File: extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/OutputSoapAttachmentsInterceptor.java (hunk 2 above)
Comment (juandesi) [inline extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/OutputSoapAttachmentsInterceptor.java]: there is another message in the class.

File: extensions/wsc/src/main/java/org/mule/extension/ws/internal/WscConnectionProvider.java (hunk 3 above)
Comment (juandesi) [inline extensions/wsc/src/main/java/org/mule/extension/ws/internal/WscConnectionProvider.java]: nope, this is all client configuration and it's directly related with the service you want to connect to.

File: extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/MessageDispatcherInterceptor.java (hunk 1 above)
Comment (juandesi) [inline extensions/wsc/src/main/java/org/mule/extension/ws/internal/interceptor/MessageDispatcherInterceptor.java]: the method has the ticket, this is one single part of it.

**Pull request merged**